docs/config.md

```diff
@@ -2,30 +2,185 @@
-| `major_version_zero`       | `bool` | `false`                     | When true, breaking changes on a `0.x` will remain as a `0.x` version. On `false`, a breaking change will bump a `0.x` version to `1.0`. [major-version-zero]                                                                                        |
-| `prerelease_offset`        | `int`  | `0`                         | In special cases it may be necessary that a prerelease cannot start with a 0, e.g. in an embedded project the individual characters are encoded in bytes. This can be done by specifying an offset from which to start counting. [prerelease-offset] |
-| `version_type`             | `str`  | `pep440`                    | Select a version type from the following options [`pep440`, `semver`]. Useful for non-python projects. [See more][version_type]                                                                                                                      |
-
-## pyproject.toml or .cz.toml
+### `name`
+
+Type: `str`
+
+Default: `"cz_conventional_commits"`
+
+Name of the committing rules to use
+
+### `version`
+
+Type: `str`
+
+Default: `None`
+
+Current version. Example: "0.1.2". Required if you use `provider = "commitizen"`.
+
+### `version_files`
+
+Type: `list`
+
+Default: `[ ]`
+
+Files were the version will be updated. A pattern to match a line, can also be specified, separated by `:` [Read more][version_files]
+
+### `version_provider`
+
+Type: `str`
+
+Default: `commitizen`
+
+Version provider used to read and write version [Read more](#version-providers)
+
+### `tag_format`
+
+Type: `str`
+
+Default: `None`
+
+Format for the git tag, useful for old projects, that use a convention like `"v1.2.1"`. [Read more][tag_format]
+
+### `update_changelog_on_bump`
+
+Type: `bool`
+
+Default: `false`
+
+Create changelog when running `cz bump`
+
+### `gpg_sign`
+
+Type: `bool`
+
+Default: `false`
+
+Use gpg signed tags instead of lightweight tags.
+
+### `annotated_tag`
+
+Type: `bool`
+
+Default: `false`
+
+Use annotated tags instead of lightweight tags. [See difference][annotated-tags-vs-lightweight]
+
+### `bump_message`
+
+Type: `str`
+
+Default: `None`
+
+Create custom commit message, useful to skip ci. [Read more][bump_message]
+
+### `allow_abort`
+
+Type: `bool`
+
+Default: `false`
+
+Disallow empty commit messages, useful in ci. [Read more][allow_abort]
+
+### `changelog_file`
+
+Type: `str`
+
+Default: `CHANGELOG.md`
+
+filename of exported changelog
+
+### `changelog_incremental`
+
+Type: `bool`
+
+Default: `false`
+
+Update changelog with the missing versions. This is good if you don't want to replace previous versions in the file. Note: when doing `cz bump --changelog` this is automatically set to `true`
```

Comment:
Will it be a bit confusing as the default behavior for `cz bump --changelog` and `cz changelog` are different?

Reply:
This has always been the case. `cz bump --changelog` avoid messing with your changelog (this allows users to modify further the file). With `cz changelog` the whole file will be recreated. An alternative would be to set `changelog_incremental` as default = True. This way `cz bump --changelog` and `cz changelog` would behave the same. But it's another breaking change

Reply:
I like `changelog_incremental=True` as default a bit more, but would be ok if we keep it as it is